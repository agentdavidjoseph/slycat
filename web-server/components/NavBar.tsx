'use strict';
import * as React from 'react';

/**
 * @param {navNames} [] of strings to be displayed
 * @param {selectedNameIndex} index of navNames[] to highlight in grey
 */
export interface NavBarProps { 
  navNames:string[]
  selectedNameIndex:Number
}

/**
 * not used
 */
export interface NavBarState {
}
/**
 * class that creates a Navbar for use in tracking progress through say a wizard or
 * some other process
 */
export default class NavBar extends React.Component<NavBarProps, NavBarState> {
  /**
   * not used
   */
  public constructor(props:NavBarProps) {
    super(props)
    this.state = {}
  }

  /**
   * returns a list of nav-items for display
   * also highlights the nav-item from selectedNameIndex
   *
   * @memberof NavBar
   */
  private getNavItemsJSX = () => {
    const NavItemsJSX = this.props.navNames.map((navName, i) => {
      return (
        <li key={i} className={this.props.selectedNameIndex === i? "nav-item active": "nav-item"}>
          <a className="nav-link">{navName}</a>
        </li>
      )
    });
    return NavItemsJSX;
  }

  public render () {
    return (
      <ul className="nav nav-pills">
        {this.getNavItemsJSX()}
      </ul>
    )
  }
}
