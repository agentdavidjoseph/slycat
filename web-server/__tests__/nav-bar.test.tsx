
import React from 'react';
import NavBar, {NavBarProps, NavBarState} from "components/NavBar";
import { mount } from "enzyme";


describe('when loading a nav bar',() =>{ 
  const names = ['test1','test2','test3'];
  const properties: NavBarProps = {
    navNames:names,
    selectedNameIndex:1
  };
  const render = mount(
    <NavBar
      {...properties}
    />
  );
  test('we have expected props on initial load', () => {
    const props = render.props() as NavBarProps;
    expect(props).toMatchSnapshot();
  });

  test('we no state', () => {
    const state = render.state() as NavBarState;
    expect(state).toMatchSnapshot();
  });

  test('we should have a full initialization', () => {
    expect(render.instance()).toMatchSnapshot();
  });
});

export default undefined