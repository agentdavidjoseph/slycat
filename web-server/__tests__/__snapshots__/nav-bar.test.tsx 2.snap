// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading a nav bar we have expected props on initial load 1`] = `
Object {
  "navNames": Array [
    "test1",
    "test2",
    "test3",
  ],
  "selectedNameIndex": 1,
}
`;

exports[`when loading a nav bar we no state 1`] = `Object {}`;

exports[`when loading a nav bar we should have a full initialization 1`] = `
<NavBar
  navNames={
    Array [
      "test1",
      "test2",
      "test3",
    ]
  }
  selectedNameIndex={1}
>
  <ul
    className="nav nav-pills"
  >
    <li
      className="nav-item"
      key="0"
    >
      <a
        className="nav-link"
      >
        test1
      </a>
    </li>
    <li
      className="nav-item active"
      key="1"
    >
      <a
        className="nav-link"
      >
        test2
      </a>
    </li>
    <li
      className="nav-item"
      key="2"
    >
      <a
        className="nav-link"
      >
        test3
      </a>
    </li>
  </ul>
</NavBar>
`;
